addons/html_builder/static/src/plugins/background_option/background_shape_option_plugin.js

```diff
@@ -328,6 +356,194 @@ export class BackgroundShapeOptionPlugin extends Plugin {
+     * be computed.
+     * @param {String} shapeName - Identifier of the selected shape.
+     * @param {Boolean} considerSelectedColor - If the `selectedColor`
+     * properties has to be taken into account in the computation of the
+     * background shape color.
+     * @returns {Object} - The computed shape color.
+     */
+    getComputedConnectionsColors(editingElement, shapeName, considerSelectedColor = true) {
+        if (
+            !shapeName.includes("html_builder/Connections/") ||
+            !editingElement.matches("[data-snippet]") ||
+            editingElement.parentElement.closest("[data-snippet]")
+        ) {
+            return {};
+        }
+        const selectedBackgroundUrl = this.getShapeStyleUrl(shapeName);
+        const defaultColors = this.getShapeDefaultColors(selectedBackgroundUrl);
+        const defaultKey = Object.keys(defaultColors)[0];
+        const shapeData = this.getShapeData(editingElement);
+        if (considerSelectedColor && shapeData.selectedColor) {
+            return {
+                [defaultKey]: normalizeColor(
+                    Object.values(shapeData.colors)[0],
+                    getHtmlStyle(this.document)
+                ),
+            };
+        }
+        const neighborEl = this.getAdjacentEl(editingElement);
+        const computedHexColor = neighborEl
+            ? this.getPerceivedBackgroundColor(neighborEl)
+            : Object.values(defaultColors)[0];
+        const curBgHexColor = this.getPerceivedBackgroundColor(editingElement);
+
+        return {
+            [defaultKey]:
+                curBgHexColor !== computedHexColor
+                    ? computedHexColor
+                    : this.getContrastingColor(computedHexColor),
+        };
+    }
+
+    /**
+     * Returns the adjacent element of a background shape. The element is either
+     * a snippet, the navbar of the header (element that has the background
+     * color of the header) or the footer.
+     * @param {HTMLElement} editingElement - The snippet that has the background
+     * shape.
+     * @returns {HTMLElement|null} - The adjacent element.
+     */
+    getAdjacentEl(editingElement) {
+        const shapeData = this.getShapeData(editingElement);
+        const isYFlipped = shapeData.flip.includes("y");
+        const elementSibling = isYFlipped ? "previousElementSibling" : "nextElementSibling";
+        let siblingEl = editingElement[elementSibling];
+        while (siblingEl) {
+            if (siblingEl.matches("[data-snippet]") && isVisible(siblingEl)) {
+                return siblingEl;
+            }
+            siblingEl = siblingEl[elementSibling];
+        }
+        const navSelector = "header nav";
+        const headerOrFooterEl = isYFlipped
+            ? this.editable.querySelector(`${navSelector}:not(.o_header_overlay ${navSelector})`)
+            : this.editable.querySelector("footer");
+        return isVisible(headerOrFooterEl) ? headerOrFooterEl : null;
```

Comment:
Maybe not such a good idea to take the header and the footer into account, to discuss